cfgov/tccp/jinja2/tccp/card.html

```diff
@@ -522,6 +522,123 @@ <h2>Cash advances</h2>
     {% else %}
         <p>This card does not offer cash advances.</p>
     {% endif %}
+    <h2>Fees</h2>
+    <dl class="m-list m-list__card-details">
+        {% if not card.periodic_fee_type %}
+            <dt>Annual fee</dt>
+            <dd>None</dd>
+        {% endif %}
+        {% if card.annual_fee %}
+            <dt>Annual fee</dt>
+            <dd>{{ '$' ~ card.annual_fee }}</dd>
+        {% endif %}
+        {% if card.monthly_fee %}
+            <dt>Monthly fee</dt>
+            <dd>{{ '$' ~ card.monthly_fee }}</dd>
+        {% endif %}
+        {% if card.weekly_fee %}
+            <dt>Weekly fee</dt>
+            <dd>{{ '$' ~ card.weekly_fee }}</dd>
+        {% endif %}
+        {% if card.other_periodic_fee_amount %}
+            <dt>{{ card.other_periodic_fee_name }}</dt>
+            <dd>{{ '$' ~ card.other_periodic_fee_amount ~ ', ' ~ card.other_periodic_fee_frequency | lower }}</dd>
+        {% endif %}
+        {% if card.fee_varies %}
+            <dt>Fee varies?</dt>
+            <dd>
+                {{ '$' ~ card.periodic_min ~ '–'  ~ card.periodic_max ~
+                    ' (' ~ card.fee_explanation | lower ~ ')'
+                }}
+            </dd>
+        {% endif %}
+        <dt>Late fee</dt>
+        <dd>
+            {% if card.late_fees %}
+                {{ '$' ~ card.late_fee_dollars ~ ' for the first late payment' if
+                    card.late_fee_dollars
+                }}
+                {{ 'and' if card.late_fee_dollars
+                    and card.late_fee_six_month_billing_cycle
+                }}
+                {{ '$' ~ card.late_fee_six_month_billing_cycle ~
+                    ' for a late payment within six billing cycles of another
+                    late payment' if card.late_fee_six_month_billing_cycle
+                }}
+                {# A handful of cards specify neither a first nor repeat late
+                    fee and put all late fee info in the late_fee_policy_details
+                    field. For cards like that, we'll show the details here and
+                    not show the "Late fee policy details" item.
+                #}
+                {{ card.late_fee_policy_details if card.late_fee_types ==
+                    ['3. If you charge late fees that are not fixed dollar amounts, please explain your late fee policy here.']
```

Comment:
It'd be better not to have all of these hardcoded strings here. Once this design is finalized we should go back and find a way to express these more concisely in the template without having to duplicate the strings verbatim.

Reply:
I added a TODO to refactor these in 6d29b89. I tried a few other ways to structure that logic without the exact string, but they all ended up being more complex than just using the exact string. I like waiting until we have all the data on the page before refactoring—there's an almost identical situation that came up in a different fee, so patterns are emerging the more data we get on the page.